### PR TITLE
Fix/NFER-3/navigation in menu

### PR DESCRIPTION
**Pull request by bziondik**
Related to: https://oat-sa.atlassian.net/browse/NFER-3

Added option in menu config `navType` : by default `'fromLast'`, also `'fromFirst'`
In case of the menu have `active` item - hoverIndex will be set on it
In case of the menu doesn't have`active` item:
for `navType: 'fromLast'` - focus on button and then using UP go to last item, when press DOWN on last item menu will be closed,
for `navType: 'fromFirst'` - focus on button and then using DOWN go to first item, when press UP on first item menu will be closed

Related PR: https://github.com/oat-sa/extension-tao-nfer/pull/169

### CODE REVIEW

**Comment by AntonTsymuk** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Comment by AntonTsymuk** [comment]
At the moment behaviour only applied to contrasts menu. Should it be applied to other menus?

**Comment by AntonTsymuk** [comment]
I am not sure that using events is a good way to update component config. Could you please discuss alternatives with @jsconan

File: src/ui/toolbox/menu.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ var menuComponentApi = {
      */
     initMenu: function initMenu() {
         this.id = this.config.control;
+        this.typeNav = this.config.navType ? this.config.navType : 'fromLast';
```
Comment (AntonTsymuk) [inline src/ui/toolbox/menu.js]: i think `navType` is more suitable variable name. Also could you please add some documentation for this configuration option

File: src/plugins/tools/itemThemeSwitcher/itemThemeSwitcher.js [hunk 2]
```diff
@@ -239,7 +239,10 @@ export default pluginFactory({
                 if (self.getState('enabled') !== false) {
                     self.menuButton.toggleMenu();
                 }
-            });
+            })
+            .on('set-themeswitcher-navtype', function(type) {
```
[the remark this answers is not in the document]
Comment (bziondik) [reply src/plugins/tools/itemThemeSwitcher/itemThemeSwitcher.js]: fixed

File: src/ui/toolbox/menu.js [hunk 3]
```diff
@@ -366,6 +399,16 @@ var menuComponentApi = {
...
             }
         }
+    },
+
+    /**
+     * Set navigation type
+     * @param {String} type - 'fromLast', 'fromFirst'
+     */
+    setNavigationType: function setItemNavigation(type) {
+        if (_.contains(['fromLast', 'fromFirst'], type)) {
```
[the remark this answers is not in the document]
Comment (bziondik) [reply src/ui/toolbox/menu.js]: fixed

File: src/ui/toolbox/menu.js [hunk 4]
```diff
@@ -366,6 +399,16 @@ var menuComponentApi = {
                 document.activeElement.blur();
             }
         }
+    },
+
+    /**
+     * Set navigation type
+     * @param {String} type - 'fromLast', 'fromFirst'
+     */
+    setNavigationType: function setItemNavigation(type) {
```
[the remark this answers is not in the document]
Comment (bziondik) [reply src/ui/toolbox/menu.js]: fixed

File: src/ui/toolbox/menu.js (hunk 1 above)
Comment (bziondik) [reply src/ui/toolbox/menu.js]: fixed

**Pull request merged**